### PR TITLE
New canine model card

### PR DESCRIPTION
**Pull request by RogerSinghChugh**
# What does this PR do?

This PR updates the model-card for the Canine model, as described in https://github.com/huggingface/transformers/issues/36979, in an attempt to standardize all model-cards.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/canine.md [hunk 1]
```diff
@@ -14,99 +14,79 @@ rendered properly in your Markdown viewer.
...
-  details for this can be found in the paper.
-
-Model checkpoints:
+# CANINE
 
-  - [google/canine-c](https://huggingface.co/google/canine-c): Pre-trained with autoregressive character loss,
-    12-layer, 768-hidden, 12-heads, 121M parameters (size ~500 MB).
-  - [google/canine-s](https://huggingface.co/google/canine-s): Pre-trained with subword loss, 12-layer,
-    768-hidden, 12-heads, 121M parameters (size ~500 MB).
+[CANINE](https://huggingface.co/papers/2103.06874) is a tokenization-free Transformer — it skips the usual step of splitting text into subwords or wordpieces and instead processes text character by character. That means it works directly with raw Unicode, making it especially useful for languages with complex or inconsistent tokenization rules, or even noisy inputs like typos. Since working with characters means handling longer sequences, CANINE uses a smart trick: it compresses the input early on (called downsampling) so the Transformer doesn’t have to process every character individually. This keeps things fast and efficient.
```
Comment (stevhliu) [inline docs/source/en/model_doc/canine.md]:
```suggestion
[CANINE](https://huggingface.co/papers/2103.06874) is a tokenization-free Transformer, It skips the usual step of splitting text into subwords or wordpieces and processes text character by character. That means it works directly with raw Unicode, making it especially useful for languages with complex or inconsistent tokenization rules and even noisy inputs like typos. Since working with characters means handling longer sequences, CANINE uses a smart trick. It compresses the input early on (called downsampling) so the transformer doesn’t have to process every character individually. This keeps things fast and efficient.
```

File: docs/source/en/model_doc/canine.md [hunk 2]
```diff
@@ -14,99 +14,79 @@ rendered properly in your Markdown viewer.
...
 
+You can find all the original CANINE checkpoints under the [Google](https://huggingface.co/google?search_models=canine) organisation.
```
Comment (stevhliu) [inline docs/source/en/model_doc/canine.md]:
```suggestion
You can find all the original CANINE checkpoints under the [Google](https://huggingface.co/google?search_models=canine) organization.
```

File: docs/source/en/model_doc/canine.md [hunk 3]
```diff
@@ -14,99 +14,79 @@ rendered properly in your Markdown viewer.
...
+The example below demonstrates how to generate embeddings with [`Pipeline`], [`AutoModel`], and from the command line.
 
-```python
->>> from transformers import CanineModel
->>> import torch
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
->>> model = CanineModel.from_pretrained("google/canine-c")  # model pre-trained with autoregressive character loss
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/canine.md:37]: Simplify it to something like this:

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="feature-extraction",
    model="google/canine-c",
    device=0
)
pipeline("Plant create energy through a process known as photosynthesis.")
```

File: docs/source/en/model_doc/canine.md [hunk 4]
```diff
@@ -14,99 +14,79 @@ rendered properly in your Markdown viewer.
...
+import torch
+from transformers import AutoModel
 
->>> model = CanineModel.from_pretrained("google/canine-c")
->>> tokenizer = CanineTokenizer.from_pretrained("google/canine-c")
+model = AutoModel.from_pretrained("google/canine-c")
 
->>> inputs = ["Life is like a box of chocolates.", "You never know what you gonna get."]
->>> encoding = tokenizer(inputs, padding="longest", truncation=True, return_tensors="pt")
+text = "Hello World!"
```
Comment (stevhliu) [inline docs/source/en/model_doc/canine.md]:
```suggestion
text = "Plant create energy through a process known as photosynthesis."
```

File: docs/source/en/model_doc/canine.md [hunk 5]
```diff
@@ -14,99 +14,79 @@ rendered properly in your Markdown viewer.
...
 
-- [Text classification task guide](../tasks/sequence_classification)
-- [Token classification task guide](../tasks/token_classification)
-- [Question answering task guide](../tasks/question_answering)
-- [Multiple choice task guide](../tasks/multiple_choice)
+```bash
+echo -e "Hello World!" | transformers-cli run \
+  --task feature-extraction \
+  --model google/canine-c \
+  --device 0
```
Comment (stevhliu) [inline docs/source/en/model_doc/canine.md]:
```suggestion
echo -e "Plant create energy through a process known as photosynthesis." | transformers-cli run --task feature-extraction --model google/canine-c --device 0
```

File: docs/source/en/model_doc/canine.md [hunk 6]
```diff
@@ -14,99 +14,79 @@ rendered properly in your Markdown viewer.
...
+```
+
+</hfoption>
+</hfoptions>
+
+## Notes
+
+- CANINE skips tokenization entirely — it works directly on raw characters, not subwords. You can use it with or without a tokenizer (e.g., CanineTokenizer is available if needed for consistency or padding).
```
Comment (stevhliu) [inline docs/source/en/model_doc/canine.md]:
```suggestion
- CANINE skips tokenization entirely — it works directly on raw characters, not subwords. You can use it with or without a tokenizer. For batched inference and training, it is recommended to use the tokenizer to pad and truncate all sequences to the same length.

    ```py
    from transformers import AutoTokenizer, AutoModel
    
    tokenizer = AutoTokenizer("google/canine-c")
    inputs = ["Life is like a box of chocolates.", "You never know what you gonna get."]
    encoding = tokenizer(inputs, padding="longest", truncation=True, return_tensors="pt")
    ```
```

File: docs/source/en/model_doc/canine.md [hunk 7]
```diff
@@ -14,99 +14,79 @@ rendered properly in your Markdown viewer.
...
+- The model supports sequences up to 2048 characters by default.
+- Classification can be done by placing a linear layer on top of the final hidden state of the special [CLS] token
+  (which has a predefined Unicode code point). For token classification tasks however, the downsampled sequence of
+  tokens needs to be upsampled again to match the length of the original character sequence (which is 2048). The
+  details for this can be found in the paper.
```
Comment (stevhliu) [inline docs/source/en/model_doc/canine.md]:
```suggestion
- CANINE is primarily designed to be fine-tuned on a downstream task. The pretrained model can be used for either masked language modeling or next sentence prediction.
```

**Review by stevhliu — COMMENTED** [review]
Thanks for your contribution!

**Comment by RogerSinghChugh** [comment]
Done with the changes @stevhliu.
Thank you!

**Review by stevhliu — APPROVED** [review]
Awesome, thanks!
